src/core/server/http/http_server.test.ts

```diff
@@ -987,9 +990,199 @@ describe('body options', () => {
     await supertest(innerServer.listener).post('/').send({ test: 1 }).expect(200, {
       parse: false,
       maxBytes: 1024, // hapi populates the default
+      timeout: 10000, // hapi populates the default
       output: 'data',
     });
   });
+
+  test('should accept a "timeout" in the body with "timeout" set to "false"', async () => {
+    const { registerRouter, server: innerServer } = await server.setup(config);
+
+    const router = new Router('', logger, enhanceWithContext);
```

Comment:
The border between testing our API and testing hapi's underlying implementation has always been thin, but would be great to test that the timeouts options are actually effective instead of just testing that `req.route.options` is correctly populated?

It would probably have to more to `src/core/server/http/integration_tests/router.test.ts` though.

Reply:
Will look into this Monday morning, but I don't know if there's an easy way to validate the different timeouts without keeping the tests running a while or making them flakey (and then they get skipped) which I don't want to do. I might be able to make a few simple cases work such as configuring the route to timeout quickly but will update if I can or can't here.

Reply:
>  if there's an easy way to validate the different timeouts without keeping the tests running a while or making them flakey 

You could configure timeouts to be small. We already use this approach and it works fine https://github.com/restrry/kibana/blob/87ca75e569d14f5489990044d2479a6f04192ab1/src/core/server/http/http_tools.test.ts#L84

Reply:
Oh cool, thanks! This example should help me out. I will see if I can add an end to end test for this using the same technique.